modules/bm_sim/include/bm_sim/match_tables.h

```diff
@@ -232,6 +233,7 @@ class MatchTable : public MatchTableAbstract {
       const std::string &name,
       p4object_id_t id,
       size_t size, const MatchKeyBuilder &match_key_builder,
+      LookupStructureFactory * lookup_factory,
```

Comment:
I believe this will go away if the lookup_factory is made a static member of `MatchUnitGeneric` as I suggested above.